packages/muze/src/canvas/component-wrapper-maker.js

```diff
@@ -253,11 +300,29 @@ const gridLayoutWrapper = (layoutManager, renderDetails, grid) => {
  * @param {Object} renderDetails Extra details required for rendering the headers
  * @return {Instance} Returns the respective wrappers for each component
  */
-export const componentWrapperMaker = (layoutManager, grid, renderDetails) => ({
-    title: createHeaderWrapper(TITLE, layoutManager, renderDetails),
-    subtitle: createHeaderWrapper(SUB_TITLE, layoutManager, renderDetails),
-    legend: createLegendWrapper(layoutManager, renderDetails),
-    grid: gridLayoutWrapper(layoutManager, renderDetails, grid),
-    verticalScrollBar: createScrollBarWrapper(VERTICAL, layoutManager, renderDetails, grid),
-    horizontalScrollBar: createScrollBarWrapper(HORIZONTAL, layoutManager, renderDetails, grid)
-});
+export const componentWrapperMaker = (canvas, grid, renderDetails) => {
+    const { rows, columns, values } = renderDetails.components;
+    const layoutManager = canvas._layoutManager;
+    const headers = {
+        title: createHeaderWrapper(TITLE, layoutManager, renderDetails),
+        subtitle: createHeaderWrapper(SUB_TITLE, layoutManager, renderDetails)
+    };
+    if (!rows && !columns && !values) {
```

Comment:
Dont use conditions like these. It's not scalable.
It's possible that other conditions might affect the drawing of the message components. Make a composite conditions based on which this will work.